atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/castor/shared/CastorShared.kt

```diff
@@ -450,25 +446,58 @@ internal class CastorShared {
             return coreProperties
                 .filterIsInstance<DIDDocument.Authentication>()
                 .flatMap { it.verificationMethods.toList() }
-                .mapNotNull {
-                    it.publicKeyMultibase?.let { publicKey ->
-                        when (DIDDocument.VerificationMethod.getCurveByType(it.type)) {
-                            Curve.SECP256K1 -> {
-                                Secp256k1PublicKey(publicKey.encodeToByteArray())
-                            }
-
-                            Curve.ED25519 -> {
-                                Ed25519PublicKey(publicKey.encodeToByteArray())
-                            }
-
-                            Curve.X25519 -> {
-                                X25519PublicKey(publicKey.encodeToByteArray())
-                            }
+                .mapNotNull { verificationMethod ->
+                    when {
+                        verificationMethod.publicKeyJwk != null -> {
+                            extractPublicKeyFromJwk(verificationMethod.publicKeyJwk)
+                        }
+
+                        verificationMethod.publicKeyMultibase != null -> {
+                            extractPublicKeyFromMultibase(verificationMethod.publicKeyMultibase, verificationMethod.type)
                         }
+
+                        else -> null
                     }
                 }
         }
 
+        private fun extractPublicKeyFromJwk(jwk: Map<String, String>): PublicKey? {
+            if (jwk.containsKey("x") && jwk.containsKey("crv")) {
+                val x = jwk["x"]
+                val crv = jwk["crv"]
+                return when (DIDDocument.VerificationMethod.getCurveByType(crv!!)) {
+                    Curve.SECP256K1 -> {
+                        Secp256k1PublicKey(x!!.encodeToByteArray())
+                    }
+
+                    Curve.ED25519 -> {
+                        Ed25519PublicKey(x!!.encodeToByteArray())
+                    }
+
+                    Curve.X25519 -> {
+                        X25519PublicKey(x!!.encodeToByteArray())
+                    }
+                }
+            }
+            return null
+        }
```

Comment:
this doesn't look correct to me
the value in `x` is dependent on the `kty` (keyType)
for example,  `kty: "EC"`  : `x` is the public key x coord

Reply:
After a discussion with @elribonazo about this, `EC` (elliptic curve) covers all the key types but does not define the curve required for each key.  Makes more sense to use the curve to create the public key.
If we need to clarify this might be a good idea to have an open chat with the team.